src/Error/Debugger.php

```diff
@@ -920,53 +920,23 @@ public static function getType($var): string
     public static function printVar($var, array $location = [], ?bool $showHtml = null): void
     {
         $location += ['file' => null, 'line' => null];
-        $file = $location['file'];
-        $line = $location['line'];
-        $lineInfo = '';
-        if ($file) {
-            $search = [];
-            if (defined('ROOT')) {
-                $search = [ROOT];
-            }
-            if (defined('CAKE_CORE_INCLUDE_PATH')) {
-                array_unshift($search, CAKE_CORE_INCLUDE_PATH);
-            }
-            $file = str_replace($search, '', $file);
-        }
-        $html = <<<HTML
-<div class="cake-debug-output" style="direction:ltr">
-%s
-<pre class="cake-debug">
-%s
-</pre>
-</div>
-HTML;
-        $text = <<<TEXT
-%s
-########## DEBUG ##########
-%s
-###########################
-
-TEXT;
-        $template = $html;
-        if ((PHP_SAPI === 'cli' || PHP_SAPI === 'phpdbg') || $showHtml === false) {
-            $template = $text;
-            if ($file && $line) {
-                $lineInfo = sprintf('%s (line %s)', $file, $line);
-            }
+        if ($location['file']) {
+            $location['file'] = static::trimPath((string)$location['file']);
         }
-        if ($showHtml === null && $template !== $text) {
-            $showHtml = true;
+
+        $debugger = static::getInstance();
+        $restore = null;
+        if ($showHtml !== null) {
+            $restore = $debugger->getConfig('exportFormatter');
+            $debugger->setConfig('exportFormatter', $showHtml ? HtmlFormatter::class : TextFormatter::class);
```

Comment:
The default here is a little confusing. Can you better explain what $showHtml = null does in the docblock?

Reply:
Sure thing.